ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -522,6 +524,64 @@ public void shouldReturn5xxOnStatementSystemError() {
     EasyMock.verify(ksqlEngine);
   }
 
+  @Test
+  public void shouldFailIfReachedActivePersistentQueriesLimit() {
+    // Given:
+    givenKsqlConfigWith(ImmutableMap.<String, Object>builder()
+        .put(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG, 3)
+        .build());
+    final String ksqlString = "CREATE STREAM test_explain AS SELECT * FROM test_stream;";
+    givenMockEngine(mockEngine -> {
+      EasyMock.expect(mockEngine.parseStatements(EasyMock.anyString()))
```

Comment:
If you switch to Mockito this statement could be part of `givenMockEngine`.

Reply:
Thoughts? It's only suggestion: it's OK to say you'd rather not / don't have time / next time / separate PR.